src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs

```diff
@@ -43,12 +43,30 @@ protected override void CleanupTest ()
 				}
 			}
 
+			UnInstallTestApp ();
+
 			ClearAdbLogcat ();
 
 
 			base.CleanupTest ();
 		}
 
+		protected static void UnInstallTestApp ()
+		{
+			if (TestPackageName.TryGetValue (TestContext.CurrentContext.Test.ID, out string packageName))
+				RunAdbCommand ($"uninstall {packageName}");
+		}
+
+		protected static string GeneratePackageName ()
+		{
+			if (TestPackageName.TryGetValue (TestContext.CurrentContext.Test.ID, out string packageName))
```

Comment:
Since `TestPackage` is a `ConcurrentDictionary<TKey, TValue>`, shouldn't this use [`.GetOrAdd()`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=net-5.0#System_Collections_Concurrent_ConcurrentDictionary_2_GetOrAdd__0_System_Func__0__1__)?

```csharp
return TestPackageName.GetOrAdd (TestContext.CurrentContext.Test.ID, key => {
    return $"com.xamarin.test{key.Replace ("-", "")}";
});
```

Reply:
Didn't even know that method existing lol. Will make that change.